cli/flox-rust-sdk/src/providers/git.rs

```diff
@@ -466,18 +466,30 @@ impl GitCommandProvider {
         repository: impl AsRef<str>,
         push_spec: impl AsRef<str>,
         force: bool,
-    ) -> Result<(), GitCommandError> {
+    ) -> Result<(), GitRemoteCommandError> {
         let mut command = self.new_command();
         command
             .arg("push")
+            .arg("--porcelain")
             .arg(repository.as_ref())
             .arg(push_spec.as_ref());
 
         if force {
             command.arg("--force");
         }
 
-        GitCommandProvider::run_command(&mut command)?;
+        match GitCommandProvider::run_command(&mut command) {
+            Ok(_) => Ok(()),
+            Err(GitCommandError::BadExit(_, _, stderr)) if stderr.contains("DENIED") => {
+                Err(GitRemoteCommandError::AccessDenied)
```

Comment:
question nonblocking: how come you aren't bubbling the error text up? Because if it contains DENIED you know there won't be any additional helpful information?

Reply:
but i added `debug` messages to catch the entire thing at least somewhere

Reply:
Makes sense I was just curious. When you `match` on the error you return a different error rather than wrapping it with `map_err` which is different than what we usually do.